File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubePlaylistLoader.java
package com.sedmelluq.discord.lavaplayer.source.youtube;

import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface;
import com.sedmelluq.discord.lavaplayer.track.AudioPlaylist;
import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
import com.sedmelluq.discord.lavaplayer.track.AudioTrackInfo;

import java.util.function.Function;

public interface YoutubePlaylistLoader {
    void setPlaylistPageCount(int playlistPageCount);

    AudioPlaylist load(HttpInterface httpInterface, String playlistId, String selectedVideoId,
                       Function<AudioTrackInfo, AudioTrack> trackFactory);
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubePlaylistLoader.kt
package com.sedmelluq.discord.lavaplayer.source.youtube

import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface
import com.sedmelluq.discord.lavaplayer.track.AudioTrackCollection
import com.sedmelluq.discord.lavaplayer.track.AudioTrackFactory

interface YoutubePlaylistLoader {
    var playlistPageCount: Int

    fun load(
        httpInterface: HttpInterface,
        playlistId: String,
        selectedVideoId: String?,
        trackFactory: AudioTrackFactory
    ): AudioTrackCollection
}
